src/main/java/com/google/api/codegen/config/ConfigGeneratorApi.java

```diff
@@ -114,10 +134,47 @@ protected void process() throws Exception {
     return methods;
   }
 
+  private static MethodConfigGenerator createFixedValueGenerator(String key, Object value) {
+    return new AutoValue_FixedValueGenerator(key, value);
+  }
+
+  private List<Object> generateInterfacesConfig() {
+    List<Object> services = new LinkedList<Object>();
+    for (Interface service : model.getSymbolTable().getInterfaces()) {
+      Map<String, Object> serviceConfig = new LinkedHashMap<String, Object>();
+      serviceConfig.put(CONFIG_KEY_SERVICE_NAME, service.getFullName());
+      serviceConfig.put(CONFIG_KEY_COLLECTIONS, generateCollectionConfigs(service));
+      serviceConfig.putAll(RetryGenerator.generateRetryDefinitions());
+      serviceConfig.put(CONFIG_KEY_METHODS, generateMethodConfigs(service));
+      services.add(serviceConfig);
+    }
+    return services;
+  }
+
+  private Map<String, Object> generateLanguageSettings() {
+    Map<String, Object> languages = new LinkedHashMap<String, Object>();
+    String packageName = generatePackageName();
+    for (Map.Entry<String, String> kvPair : LANGUAGE_PACKAGE_SEPARATORS.entrySet()) {
+      languages.put(kvPair.getKey(), formatPackageNameEntry(packageName, kvPair.getValue()));
+    }
+    return languages;
+  }
+
+  private String generatePackageName() {
+    int index = Preconditions.checkPositionIndex(0, model.getFiles().size());
+    return model.getFiles().get(index).getFullName();
+  }
+
+  private Map<String, Object> formatPackageNameEntry(String packageName, String separator) {
+    Map<String, Object> packageNameMap = new LinkedHashMap<String, Object>();
+    packageNameMap.put(CONFIG_KEY_PACKAGE_NAME, packageName.replace(".", separator));
+    return packageNameMap;
+  }
+
   private void dump(Map<String, Object> data) throws IOException {
     DumperOptions dumperOptions = new DumperOptions();
     dumperOptions.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
-    dumperOptions.setPrettyFlow(true);
+    dumperOptions.setPrettyFlow(false);
```

Comment:
The only effect is has is on empty lists or dictionaries, so when it is false you get this:
someemptything: []
and when it is true you get this:
someemptything: [
]